server/server.go

```diff
@@ -171,6 +173,19 @@ func (s *Server) genericOAuth(logger chronograf.Logger, auth oauth2.Authenticato
 	return &gen, genMux, s.UseGenericOAuth2
 }
 
+func (s *Server) genericRedirectURL() string {
+	if s.PublicURL == "" {
+		return ""
+	}
+
+	genericName := "generic"
+	if s.GenericName != "" {
+		genericName = s.GenericName
+	}
+
+	return path.Join(s.PublicURL, s.Basepath, "oauth", genericName, "callback")
```

Comment:
oh. that's my fault. I just checked that I can redirect to oauth provider's page, didn't noticed the callback url is wrong. I'll fix it.